exec/java-exec/src/main/java/org/apache/drill/exec/server/options/SessionOptionManager.java

```diff
@@ -46,6 +47,33 @@ public SessionOptionManager(final OptionManager systemOptions, final UserSession
   }
 
   @Override
+  public void deleteOption(final String name, final OptionValue.OptionType type) {
+    if (type == OptionValue.OptionType.SESSION) {
+      try { // ensure option exists
```

Comment:
You only ensure the option exists when `type == SESSION`. How about _SYSTEM_, is it done by the underlying _fallback_ OptionManager ?